### PR TITLE
[CSM] add screenshot api lua

### PR DESCRIPTION
**Pull request by Dumbeldor**
#5394

### CODE REVIEW

File: doc/client_lua_api.md [hunk 1]
```diff
@@ -703,6 +703,8 @@ Call these functions only at load time!
 * `minetest.get_protocol_version()`
     * Returns the protocol version of the server.
     * Might not be accurate at start up as the client might not be connected to the server yet, in that case it will return 0.
+* `minetest.screenshot()`
```
Comment (nerzhul) [inline doc/client_lua_api.md]: take_screenshot

File: src/script/lua_api/l_client.cpp [hunk 2]
```diff
@@ -241,6 +241,13 @@ int ModApiClient::l_get_protocol_version(lua_State *L)
 	return 1;
 }
 
+int ModApiClient::l_take_screenshot(lua_State *L)
+{
+	Client* client = getClient(L);
```
Comment (SmallJoker) [inline src/script/lua_api/l_client.cpp]: `Client *client ...`

**Comment by sofar** [comment]
#5672

**Comment by sofar** [comment]
My bad, "back" button did something I didn't expect

**Comment by sofar** [comment]
Do we really want this? This will make it trivial to fill up someones drive with files.

**Comment by Ekdohibs** [comment]
For that a few other changes (like maybe client API that will make predictions to the world state), I think what we really want is *two* Lua APIs: one that would be for locally-installed client mods, and one for client mods sent from server. Of course, they would share most of the API calls, but locally-installed mods do not need as much security and would be allowed API calls like this one, while mods sent from server would be much more limited with these API calls, but would probably however be allowed to make calls that predict changes to the world state or that modify the player physics. Since we only have the former for the moment, I don't think this should be an objection to this PR, as clients can have that code only if they install the mod themselves.

**Comment by nerzhul** [comment]
ofc now we can do it, rate limiting seems lgtm to code

**Pull request merged**